src/convertToObject.js

```diff
@@ -1,18 +1,22 @@
 'use strict';
 
 /**
- * Implement convertToObject function:
- *
- * Function takes a string with styles (see an example in [stylesString.js](./stylesString.js))
- * and returns an object where CSS properties are keys
- * and values are the values of related CSS properties (see an exampl in [test file](./convertToObject.test.js))
  *
  * @param {string} sourceString
  *
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  return sourceString
+    .split(';')
+    .map(element => element.trim())
+    .filter(element => element.length !== 0)
+    .map(element => element.split(':'))
+    .reduce((obj, element) => {
+      return {
+        ...obj, [element[0].trim()]: element[1].trim(),
+      };
+    }, {});
```

Comment:
```suggestion
    .reduce((obj, element) => {
      const [property, value] = element.split(':');

      obj[property.trim()] = value.trim();

      return obj;
    }, {});
```
You can add a property to the object and return it, no need to create a new object and copy everything on each step. Consider using destructuring and better naming for clarity. Also, one `.map` pass can be removed.

Reply:
I guess you forgot to fix it.